client/state/happychat/chat/reducer.js

```diff
@@ -88,66 +83,70 @@ const timelineEvent = ( state = {}, action ) => {
 				files: message.files,
 				timestamp: maybeUpscaleTimePrecision( message.timestamp ),
 				user_id: message.user.id,
-				type: get( message, 'type', 'message' ),
-				links: get( message, 'meta.links' ),
+				type: message.type ?? 'message',
+				links: message.meta?.links,
 			};
+		}
 	}
 	return state;
 };
 
-const sortTimeline = ( timeline ) =>
-	sortBy( timeline, ( event ) => parseInt( event.timestamp, 10 ) );
+const getEventTimestamp = ( event ) => parseInt( event.timestamp, 10 );
+const timelineSortCmp = ( a, b ) => getEventTimestamp( a ) - getEventTimestamp( b );
+const sortTimeline = ( timeline ) => timeline.concat().sort( timelineSortCmp );
 
 /**
  * Adds timeline events for happychat
  *
  * @param  {object} state  Current state
  * @param  {object} action Action payload
  * @returns {object}        Updated state
- *
  */
 const timelineReducer = ( state = [], action ) => {
 	switch ( action.type ) {
 		case HAPPYCHAT_IO_RECEIVE_MESSAGE:
-		case HAPPYCHAT_IO_RECEIVE_MESSAGE_OPTIMISTIC:
+		case HAPPYCHAT_IO_RECEIVE_MESSAGE_OPTIMISTIC: {
 			// if meta.forOperator is set, skip so won't show to user
-			if ( get( action, 'message.meta.forOperator', false ) ) {
+			if ( action.message.meta?.forOperator ?? false ) {
```

Comment:
Can also remove `?? false` here.

Reply:
Indeed; addressed in f69ef0c75761ef72b2aa331f9a599d244028e290.